addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -434,52 +442,65 @@ export class HtmlField extends Component {
+                                '<meta charset="utf-8"/>' +
+                                '<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/>\n' +
+                                '<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no"/>\n' +
+                            '</head>\n' +
+                            '<body class="o_in_iframe o_readonly" style="overflow: hidden;">\n' +
+                                '<div id="iframe_target"></div>\n' +
+                            '</body>' +
+                            '</html>');
                 }
 
-                const iframeTarget = cwindow.document.querySelector('#iframe_target');
-                iframeTarget.innerHTML = value;
+                if (asset) {
+                    for (const cssLib of asset.cssLibs) {
+                        const link = cwindow.document.createElement('link');
+                        link.setAttribute('type', 'text/css');
+                        link.setAttribute('rel', 'stylesheet');
+                        link.setAttribute('href', cssLib);
+                        cwindow.document.head.append(link);
+                    }
+                    for (const cssContent of asset.cssContents) {
+                        const style = cwindow.document.createElement('style');
+                        style.setAttribute('type', 'text/css');
+                        const textNode = cwindow.document.createTextNode(cssContent);
+                        style.append(textNode);
+                        cwindow.document.head.append(style);
+                    }
+                }
 
-                const script = cwindow.document.createElement('script');
-                script.setAttribute('type', 'text/javascript');
-                const scriptTextNode = document.createTextNode(
-                    `if (window.top.${this._onUpdateIframeId}) {` +
-                        `window.top.${this._onUpdateIframeId}(${_avoidDoubleLoad})` +
-                    `}`
-                );
-                script.append(scriptTextNode);
-                cwindow.document.body.append(script);
+                if (!this.safeHtmlEdit) {
+                    const iframeTarget = cwindow.document.querySelector('#iframe_target');
+                    iframeTarget.innerHTML = value;
+
+                    const script = cwindow.document.createElement('script');
+                    script.setAttribute('type', 'text/javascript');
+                    const scriptTextNode = document.createTextNode(
+                        `if (window.top.${this._onUpdateIframeId}) {` +
+                            `window.top.${this._onUpdateIframeId}(${_avoidDoubleLoad})` +
+                        `}`
+                    );
+                    script.append(scriptTextNode);
+                    cwindow.document.body.append(script);
+                } else {
+                    cwindow.document.documentElement.innerHTML = value;
```

Comment:
This is fine because this is guaranteed to be in a sandboxed iframe (test also ensures this doesn't change)